pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -741,11 +741,10 @@ public void connectionOpened(final ClientCnx cnx) {
 
         boolean isDurable = subscriptionMode == SubscriptionMode.Durable;
         MessageIdData startMessageIdData = null;
-        if (isDurable) {
-            // For regular durable subscriptions, the message id from where to restart will be determined by the broker.
-            startMessageIdData = null;
-        } else if (startMessageId != null) {
-            // For non-durable we are going to restart from the next entry
+
+        // For regular durable subscriptions, the message id from where to restart will be determined by the broker.
+        // For non-durable we are going to restart from the next entry.
+        if (!isDurable && startMessageId != null) {
```

Comment:
I suggest changing `startMessageIdData` to a `final` variable so that code could be more clear and the variable cannot be assigned twice or more.

i.e.

```java
        final MessageIdData startMessageIdData;
        if (!isDurable && startMessageId != null) {
            // For non-durable we are going to restart from the next entry
            startMessageIdData = new MessageIdData()
                    .setLedgerId(startMessageId.getLedgerId())
                    .setEntryId(startMessageId.getEntryId())
                    .setBatchIndex(startMessageId.getBatchIndex());
        } else {
            startMessageIdData = null;
        }
```

Reply:
done @BewareMyPower